pkg/reconciler/v1alpha1/clusteringress/resources/gateway.go

```diff
@@ -84,14 +92,40 @@ func MakeServers(ci *v1alpha1.ClusterIngress) []v1alpha3.Server {
 				Mode:              v1alpha3.TLSModeSimple,
 				ServerCertificate: tls.ServerCertificate,
 				PrivateKey:        tls.PrivateKey,
-				// TODO(zhiminx): When Istio 1.1 is ready, set "credentialName" of TLS
-				// with the secret name.
+				CredentialName:    credentialName,
 			},
 		})
 	}
 	return sortServers(servers)
 }
 
+// GatewayServiceNamespace returns the namespace of the gateway service that the `Gateway` object
+// with name `gatewayName` is associated with.
+func GatewayServiceNamespace(ingressGateways []config.Gateway, gatewayName string) (string, error) {
+	for _, gw := range ingressGateways {
+		if gw.GatewayName != gatewayName {
+			continue
+		}
+		// serviceURL should be of the form serviceName.namespace.<domain>, for example
```

Comment:
`should` be is not strong enough, if you want to ignore the checking that slice will be at least of size 2.